addons/stock/models/stock_move.py

```diff
@@ -1210,7 +1210,7 @@ def _onchange_product_uom(self):
     def _key_assign_picking(self):
         self.ensure_one()
         keys = (self.group_id, self.location_id, self.location_dest_id, self.picking_type_id)
-        if self.partner_id and (self.location_id.usage == 'transit' or self.location_dest_id.usage == 'transit'):
+        if self.partner_id and not self.group_id and any(loc.usage in ('transit', 'supplier', 'customer') for loc in (self.location_id, self.location_dest_id)):
```

Comment:
@lase-odoo I think he meant just  `if self.partner_id and not self.group_id:`

Reply:
Oh well ^^